### PR TITLE
Port to .Net Core

### PR DESCRIPTION
**Pull request by axunonb**
Here's the ready port to .Net Core. I tried to make the 2 new solutions (.Net 4 and .Net Core) build with AppVeyor, but I got stuck, sorry.

These 7 tests are not covered in .Net Core because ```Color``` and ```DataSet``` are not supported (yet?) in .Net Core. All other tests complete successfully.
```
DeserializationTest  DesColorExampleTest
DeserializationTest  DesDataSetAndDataTableDynamicKnownTypes
KnownTypeTests       DataSetAndDataTableSerializationTest
KnownTypeTests       TestColorNames
KnownTypeTests       TestExtensionMethod
KnownTypeTests       TestSingleKnownTypeSerialization
SerializationTest    ColorExampleTest
```
Eventually ```System.Drawing.Color``` could be replaced with another known type also available in .Net Core.

### CODE REVIEW

**Comment by 304NotModified** [comment]
Are you only stuck with appveyor? I could help with that.

**Comment by axunonb** [comment]
Yeah, it's only appveyor, everything else is fine. Help is would be highly appreciated.

**Comment by 304NotModified** [comment]
Looking at it now. :)

**Comment by 304NotModified** [comment]
have you upgraded nunit?

**Comment by axunonb** [comment]
yes, because .NetCore is supported only with NUnit 3

**Comment by 304NotModified** [comment]
OK, got the build running but not yet opencover with nunit3

**Comment by 304NotModified** [comment]
the unit tests are also succeeding on .net core?

**Comment by axunonb** [comment]
Sure :)

**Comment by axunonb** [comment]
... except for the 7 tests mentioned above, which are excluded with conditional compilation.

**Comment by 304NotModified** [comment]
mmm it worked, but opencover took +4 minutes :| (for non core tests)

**Comment by 304NotModified** [comment]
YAXLibTests.SerializationTest.SerializeExceptionShouldNotThrowExceptions
YAXLibTests.dll 4 min 19 sec 64 ms

**Comment by 304NotModified** [comment]
any idea?

**Comment by axunonb** [comment]
I have a look. SerializeExceptionShouldNotThrowExceptions on .NetCore or .Net

**Comment by 304NotModified** [comment]
.NEt

**Comment by axunonb** [comment]
passed locally, like all others

**Comment by 304NotModified** [comment]
also take ages on my machine

image

**Comment by 304NotModified** [comment]
well 24 seconds. still long

**Comment by 304NotModified** [comment]
image

**Comment by axunonb** [comment]
yep, core tests are significantly more slow. SerializeExceptionShouldNotThrowExceptions takes 23 secs on my machine

**Comment by 304NotModified** [comment]
it's non-core

**Comment by 304NotModified** [comment]
Release build

**Comment by axunonb** [comment]
yes SerializeExceptionShouldNotThrowExceptions non core takes 23 secs

**Comment by 304NotModified** [comment]
it think it's a issue in the code, not the test.

image

14 seconds for one call. VS also have issues with it

**Comment by 304NotModified** [comment]
LOL nice test

image

**Comment by axunonb** [comment]
Oh... Just checked the 2.14 build (before I starred), takes about the same time here

**Comment by 304NotModified** [comment]
OK, then is a current bug

**Comment by axunonb** [comment]
Interesting enough, that in .NetCore Tests SerializeExceptionShouldNotThrowExceptions takes 100ms...

**Comment by 304NotModified** [comment]
maybe it got worse with nunit3

**Comment by 304NotModified** [comment]
this build, 50 sec

https://ci.appveyor.com/project/SinaIravanian/yaxlib/build/2.13.21

**Comment by 304NotModified** [comment]
new build, 5 minutes

https://ci.appveyor.com/project/304NotModified/yaxlib-n64ad/build/2.13.12

**Comment by axunonb** [comment]
Looks like NUnit 3 is the cause, strange. But as long as YAXLib peforms well...

**Comment by 304NotModified** [comment]
well not sure.

**Comment by 304NotModified** [comment]
I think `exceptionSerialized` is huge

**Comment by 304NotModified** [comment]
there is somewhere a loop. And that's a bug in Yaxlib. Doubting if we should disable this teset

**Comment by 304NotModified** [comment]
it also breaks my VS

**Comment by 304NotModified** [comment]
why do we need NUnit3TestAdapter btw?

**Comment by axunonb** [comment]
in order to make the VS test window work

**Comment by 304NotModified** [comment]
if we set ` ser.MaxRecursion = 10;`, it's better

**Comment by 304NotModified** [comment]
event with  ser.MaxRecursion = 10;, the result string is 18142 lines :P

**Comment by axunonb** [comment]
Just built a console app for SerializeExceptionShouldNotThrowExceptions with the new code: Release .net 4.5: 9 secs

**Comment by axunonb** [comment]
--- with  ser.MaxRecursion: 1700ms

**Comment by 304NotModified** [comment]
got these errors on .net core:

1) Error : YAXLibTests.SerializationTest.CultureChangeTest
System.Globalization.CultureNotFoundException : Culture is not supported.
Parameter name: name
fa-FR is an invalid culture identifier.
   at System.Globalization.CultureInfo..ctor(String name, Boolean useUserOverride)
   at YAXLibTests.SerializationTest.CultureChangeTest() in C:\xaxlib\YAXLibTests\SerializationTest.cs:line 129
2) Failed : YAXLibTests.SerializationTest.FormattingTest
  String lengths are both 811. Strings differ at index 212.
  Expected: "...07</ModificationDate>\r\n  <PI>3.14159</PI>\r\n  <NaturalExp>\r..."
  But was:  "...07</ModificationDate>\r\n  <PI>3,14159</PI>\r\n  <NaturalExp>\r..."
  ----------------------------------------------^
at YAXLibTests.SerializationTest.FormattingTest() in C:\xaxlib\YAXLibTests\SerializationTest.cs:line 399
Test Run Summary

File: YAXLibTests/SerializationTest.cs [hunk 1]
```diff
@@ -113,26 +117,44 @@ public void BookWithDecimalPriceTest()
...
+#else
+			Thread.CurrentThread.CurrentCulture = new CultureInfo("fr-FR");
+#endif
+			var serializer = new YAXSerializer(typeof(CultureSample), YAXExceptionHandlingPolicies.DoNotThrow, YAXExceptionTypes.Warning, YAXSerializationOptions.SerializeNullObjects);
             string frResult = serializer.Serialize(CultureSample.GetSampleInstance());
 
-            Thread.CurrentThread.CurrentCulture = CultureInfo.GetCultureInfo("fa-IR");
-            serializer = new YAXSerializer(typeof(CultureSample), YAXExceptionHandlingPolicies.DoNotThrow, YAXExceptionTypes.Warning, YAXSerializationOptions.SerializeNullObjects);
+#if FXCORE
+			CultureInfo.CurrentCulture = new CultureInfo("fa-FR");
```
Comment (304NotModified) [inline YAXLibTests/SerializationTest.cs]: this is a bug? should be fa-IR?

Comment (axunonb) [inline YAXLibTests/SerializationTest.cs]: Yes, that looks like a typo. Although the test succeeds here.

Comment (axunonb) [inline YAXLibTests/SerializationTest.cs]: The typo does not influence the test result because date formats are the same, but it was a bug.

**Comment by axunonb** [comment]
This was a good session, thanks. Will you update my PR with the appveyor changes?

**Comment by 304NotModified** [comment]
https://github.com/axunonb/YAXLib/pull/1

**Comment by 304NotModified** [comment]
yes I have to go also

**Comment by 304NotModified** [comment]
thanks for the help!

**Comment by axunonb** [comment]
thanks for yours! good night

**Comment by 304NotModified** [comment]
date time issue could be https://github.com/sinairv/YAXLib/issues/35

**Comment by 304NotModified** [comment]
Ah all test are succeeding  in .NET Core and non-core

**Comment by sinairv** [comment]
Great work guys. Thanks.

**Comment by axunonb** [comment]
See https://github.com/thangchung/awesome-dotnet-core/pull/79#issuecomment-288285456:
"... it is really awesome Serialization lib 👍"

**Comment by sinairv** [comment]
Thank you @axunonb. This is really good to make it visible to others.

**Pull request merged**